discovery-infra/download_logs.py

```diff
@@ -59,13 +59,34 @@ def min_number_of_log_files(cluster, is_controller_expected):
     else:
         return  len(cluster['hosts'])
 
-def download_logs(client: InventoryClient, cluster: dict, dest: str,
-                  must_gather: bool, retry_interval: int = RETRY_INTERVAL):
+def is_update_needed(output_folder: str, update_on_events_update: bool, client: InventoryClient, cluster: dict):
+
+    if not os.path.isdir(output_folder):
+        return True
+    if not update_on_events_update:
+        return False
+
+    destination_event_file_path = os.path.join(output_folder, f"cluster_{cluster['id']}_events.json")
+    latest_event_file_path = os.path.join("/tmp", f"cluster_{cluster['id']}_events.json")
+
+    with suppress(assisted_service_client.rest.ApiException):
+        client.download_cluster_events(cluster['id'], latest_event_file_path)
+
+    if filecmp.cmp(destination_event_file_path, latest_event_file_path):
+        os.remove(latest_event_file_path)
+        log.info("no new events found for {}".format(destination_event_file_path))
+        return False
+
+    log.info("update needed, new events found, deleting {} ".format(destination_event_file_path))
+    os.remove(destination_event_file_path)
+    os.remove(latest_event_file_path)
+    return True
 
+def download_logs(client: InventoryClient, cluster: dict, dest: str, must_gather: bool, update_by_events: bool, retry_interval: int = RETRY_INTERVAL):
     output_folder = get_logs_output_folder(dest, cluster)
 
-    if os.path.isdir(output_folder):
-        log.info(f"Skipping. The logs directory {output_folder} already exists.")
+    if not is_update_needed(output_folder, update_by_events, client, cluster,):
+        log.info(f"Skipping, no need to update {output_folder}.")
```

Comment:
Log the cluster_id

Reply:
the cluster_id is embedded in the output folder name